src/signature.rs

```diff
@@ -167,10 +170,25 @@ impl<'a> VerificationHelper for Helper<'a> {
     fn get_certs(&mut self, ids: &[openpgp::KeyHandle]) -> openpgp::Result<Vec<openpgp::Cert>> {
         let mut certs = Vec::new();
         for id in ids {
-            let matches = self.certstore.lookup_by_cert_or_subkey(id);
-            for lc in matches? {
-                certs.push(lc.to_cert()?.clone());
-            }
+            match self.certstore.lookup_by_cert_or_subkey(id) {
+                Ok(matches) => {
+                    for lc in matches {
+                        // lc.to_cert() should never fail for ephemeral SequoiaMechanism, where the in-memory cert store always creates
+                        // a LazyCert from a parsed Cert. It could fail for non-ephemeral contexts, where the LazyCert typically originates
+                        // as RawCert and the parse Cert would be created here.
+                        certs.push(lc.to_cert()?.clone());
+                    }
+                }
+                Err(e) => {
+                    if let Some(sequoia_cert_store::store::StoreError::NotFound(_)) =
+                        e.downcast_ref::<sequoia_cert_store::store::StoreError>()
```

Comment:
```suggestion
                        e.downcast_ref()
```
I think this type annotation is unnecessary as Rust should infer it.

Reply:
Thanks, done.